.github/workflows/ci-vulnerability-scans.yml

```diff
@@ -0,0 +1,103 @@
+# GitHub Actions CI workflow that runs vulnerability scans on the application's Docker image
+# to ensure images built are secure before they are deployed. 
+
+# NOTE: The workflow isn't able to pass the docker image between jobs, so each builds the image.
+#       A future PR will pass the image between the scans to reduce overhead and increase speed
+name: API Linting & Testing CI
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - app/**
+  pull_request:
+    paths:
+        - app/**
+
+jobs:
+  hadolint-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # Scans Dockerfile for any bad practices or issues
+      - name: Scan Dockerfile by hadolint
+        uses: hadolint/hadolint-action@v3.1.0
+        with:
+          dockerfile: app/Dockerfile
+          format: json
+          failure-threshold: warning
+
+  trivy-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build and tag Docker image for scanning
+        id: build-image
+        run: |
+          make release-build
+          echo "image=template-infra-app:latest" >> $GITHUB_OUTPUT
+
+      - name: Run Trivy vulnerability scan
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: "image" # Scan type, image or fs
+          image-ref: ${{ steps.build-image.outputs.image }}
+          format: "table" # Can be table, json, sarif, or github
+          exit-code: "1" # Exit code to return if any vulnerabilities are found
+          ignore-unfixed: true # Ignore unpatched/unfixed vulnerabilities
+          vuln-type: "os" # Can be os or library, depending on the type of scan you want to run
+          security-checks: "vuln,secret" # Can be any combo of vuln, secret, and config
+
+  anchore-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build and tag Docker image for scanning
+        id: build-image
+        run: |
+          make release-build
+          echo "image=template-infra-app:latest" >> $GITHUB_OUTPUT
+
+      - name: Run Anchore vulnerability scan
+        uses: anchore/scan-action@v3
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          severity-cutoff: "negligible" # Will ignore any findings that are negligible or lower
+          output-format: "table" # Can be table, json, or sarif
+
+  dockle-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build and tag Docker image for scanning
+        id: build-image
+        run: |
+          make release-build
+          echo "image=template-infra-app:latest" >> $GITHUB_OUTPUT
+
+      - name: Run Dockle container linter
+        uses: hands-lab/dockle-action@v1
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          exit-code: '1'
+          exit-level: WARN
+
+  image-scans:
```

Comment:
Is this step actually needed? It looks like it's only purpose is to echo that the tests finished

Reply:
Technically, we don't but I had it there in case there was something that needed to be ran after the jobs finished. I'll remove it in the next commit since it isn't really needed